test/unit/libraries/EnumerableGuardianMap/set.t.sol

```diff
@@ -2,16 +2,109 @@
-    function test_Set_UpdatesValuesForKeysAlreadyInTheSet() public view { }
-    function test_Set_RevertWhen_MaxNumberOfGuardiansReached() public view { }
-    function test_Set_AddsAKey() public view { }
-    function test_Set_AddsSeveralKeys() public view { }
+    function test_Set_AddsAKey() public {
+        guardiansStorage[accountAddress].set({
+            key: guardian1,
+            value: GuardianStorage(GuardianStatus.REQUESTED, guardianWeights[1])
+        });
+    }
+
+    function test_Set_AddsSeveralKeys() public {
+        bool result;
+
+        result = guardiansStorage[vm.addr(1)].set({
+            key: guardian1,
+            value: GuardianStorage(GuardianStatus.REQUESTED, guardianWeights[1])
+        });
+        assertEq(result, true);
+        result = guardiansStorage[vm.addr(2)].set({
+            key: guardian2,
+            value: GuardianStorage(GuardianStatus.REQUESTED, guardianWeights[2])
+        });
+        assertEq(result, true);
+        require(
+            guardiansStorage[vm.addr(1)]._values[guardian1].status ==
+                GuardianStatus.REQUESTED,
+            "Expected status to be REQUESTED"
+        );
+        require(
+            guardiansStorage[vm.addr(2)]._values[guardian2].status ==
+                GuardianStatus.REQUESTED,
+            "Expected status to be REQUESTED"
+        );
+        require(
+            guardiansStorage[vm.addr(3)]._values[guardian3].status ==
+                GuardianStatus.NONE,
+            "Expected status to be NONE"
+        );
+    }
+
+    function test_Set_ReturnsFalseWhen_AddingKeysAlreadyInTheSet() public {
+        bool result;
+
+        result = guardiansStorage[vm.addr(1)].set({
+            key: guardian1,
+            value: GuardianStorage(GuardianStatus.REQUESTED, guardianWeights[1])
+        });
+        assertEq(result, true);
+        result = guardiansStorage[vm.addr(1)].set({
+            key: guardian1,
+            value: GuardianStorage(GuardianStatus.REQUESTED, guardianWeights[1])
+        });
+        assertEq(result, false);
+    }
+
+    function test_Set_UpdatesValuesForKeysAlreadyInTheSet() public {
+        bool result;
+
+        result = guardiansStorage[vm.addr(1)].set({
+            key: guardian1,
+            value: GuardianStorage(GuardianStatus.REQUESTED, guardianWeights[1])
+        });
+        assertEq(result, true);
+        result = guardiansStorage[vm.addr(1)].set({
+            key: guardian1,
+            value: GuardianStorage(GuardianStatus.ACCEPTED, guardianWeights[1])
+        });
+        assertEq(result, false);
+        require(
+            guardiansStorage[vm.addr(1)]._values[guardian1].status ==
+                GuardianStatus.ACCEPTED,
+            "Expected status to be ACCEPTED"
+        );
+    }
+
+    function test_Set_RevertWhen_MaxNumberOfGuardiansReached() public {
+        bool result;
+        // TODO: Can it be acceptable to number 33?
```

Comment:
Regarding current implementation, it allows to store 33 guardians. Is that expected behavior?

Reply:
Ah yes good catch, expected behaviour should be 32. Could you change to assertion in the library to ensure that the max number is 32?

Reply:
@JohnGuilding Fixed.